src/components/src/hooks/use-fetch-vector-tile-metadata.ts

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: MIT
+// Copyright contributors to the kepler.gl project
+
+import {useCallback, useEffect, useState} from 'react';
+
+import {MVTSource, TileJSON} from '@loaders.gl/mvt';
+import {PMTilesSource, PMTilesMetadata} from '@loaders.gl/pmtiles';
+
+import {VectorTileMetadata, VectorTileType} from '@kepler.gl/layers';
+
+type FetchVectorTileMetadataProps = {
+  url: string | null;
+  type: VectorTileType;
+  process?: (json: PMTilesMetadata | TileJSON) => VectorTileMetadata | Error | null;
+};
+
+const DEFAULT_PROCESS_FUNCTION = (json: PMTilesMetadata | TileJSON): VectorTileMetadata => {
+  return {
+    metaJson: null,
+    bounds: null,
+    center: null,
+    maxZoom: null,
+    minZoom: null,
+    fields: [],
+    ...json
+  };
+};
+
+type FetchVectorTileMetadataReturn = {
+  data: VectorTileMetadata | null;
+  loading: boolean;
+  error: Error | null;
+};
+
+/** Hook to fetch and return mvt or pmtiles metadata. */
+export default function useFetchVectorTileMetadata({
+  type,
+  url,
+  process = DEFAULT_PROCESS_FUNCTION
+}: FetchVectorTileMetadataProps): FetchVectorTileMetadataReturn {
+  const [error, setError] = useState<Error | null>(null);
+  const [data, setData] = useState<VectorTileMetadata | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const setProcessedData = useCallback(
+    (value: PMTilesMetadata | TileJSON | null) => {
+      if (!value) {
+        return setError(value);
+      }
+      const processedData = process(value);
+      if (processedData instanceof Error) {
+        setError(processedData);
+      } else {
+        setData(processedData);
+      }
+    },
+    [setError, setData, process]
+  );
+
+  useEffect(() => {
+    const getAndProcessMetadata = async () => {
+      setError(null);
+      setData(null);
+      if (url) {
+        setLoading(true);
+
+        try {
+          const tileSource =
+            type === VectorTileType.REMOTE
```

Comment:
Is this stuff needed in kepler.gl? Does it make sense to provide support for LOCAL files in kepler? Just a rhetorical question, I'm not against it, if it makes it easier to integrate kepler into platforms.